sinatra-contrib/lib/sinatra/runner.rb

```diff
@@ -121,26 +127,29 @@ def ping(timeout=30)
     def alive?
       3.times { get(ping_path) }
       true
-    rescue Errno::ECONNREFUSED, Errno::ECONNRESET, EOFError, SystemCallError, OpenURI::HTTPError, Timeout::Error
+    rescue EOFError, SystemCallError, OpenURI::HTTPError, Timeout::Error
```

Comment:
Rubocop was reporting a ShadowedException. It seems SystemCallError would also rescue Errno::ECONNREFUSED, Errno::ECONNRESET errors.